pennylane/templates/embeddings.py

```diff
@@ -93,28 +93,41 @@ def AmplitudeEmbedding(features, wires, pad=False, normalize=False):
         normalize (Boolean): controls the activation of automatic normalization
     """
 
-    if not isinstance(wires, Iterable):
-        raise ValueError("Wires needs to be a list of wires that the embedding uses; got {}.".format(wires))
+    if isinstance(wires, int):
+        wires = [wires]
+
+    features = np.array(features)
 
     n_features = len(features)
     n_amplitudes = 2**len(wires)
 
     if n_amplitudes < n_features:
-        raise ValueError("AmplitudeEmbedding requires the size of feature vector to be smaller than or equal to 2**len(wires), which is {}; "
+        raise ValueError("AmplitudeEmbedding requires the size of feature vector to be "
+                         "smaller than or equal to 2**len(wires), which is {}; "
                          "got {}.".format(n_amplitudes, n_features))
 
     if pad and n_amplitudes >= n_features:
         features = np.pad(features, (0, n_amplitudes-n_features), 'constant')
 
-    if  not pad and n_amplitudes != n_features:
-        raise ValueError("AmplitudeEmbedding with no padding requires a feature vector of size 2**len(wires), which is {}; "
-                         "got {}.".format(n_amplitudes, n_features))
+    if not pad and n_amplitudes != n_features:
+        raise ValueError("AmplitudeEmbedding must get a feature vector of size 2**len(wires), "
+                         "which is {}; got {}. Use ``pad=True`` to automatically pad the "
+                         "features with zeros.".format(n_amplitudes, n_features))
 
-    if normalize and np.linalg.norm(features, 2) != 1:
-        features = features * (1/np.linalg.norm(features, 2))
+    # Get normalization
+    norm = 0
+    for f in features:
+        if type(f) is Variable:
```

Comment:
Note this line is currently causing tests to fail

Reply:
I see...only ran the templates tests, but configuration fails.

I need to rethink this. We had trouble with in-template-checks before. Do you have an idea how to cater for arrays, lists, variables, tensors etc at the same time? 

I start thinking that we just do not use any checks in templates, since it is strictly speaking inside a quantum function that should only do quantum operations...

Reply:
> I start thinking that we just do not use any checks in templates, since it is strictly speaking inside a quantum function that should only do quantum operations...
I agree :+1:

Reply:
For now I will leave as is, and we can think about a general strategy elsewhere.

Reply:
The test passes now, except from a version error in the about() function that is also in master.